File: MINIMUM-SIZE-SUBARRAY-SUM/msss.py
from typing import List

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = csum = 0
        minlen = float('inf')
        
        for right in range(len(nums)):
            csum += nums[right]

            while csum >= target:
                minlen = min(minlen, right - left + 1)
                csum -= nums[left]
                left += 1

        return 0 if minlen == float('inf') else minlen

if __name__ == "__main__":
    s = Solution()
    # print(s.minSubArrayLen(7, [2,3,1,2,4,3]))  # Output: 2
    # print(s.minSubArrayLen(4, [1,4,4]))         # Output: 1
    print(s.minSubArrayLen(11, [1,1,1,1,1,1,1]))  # Output: 0
    # print(s.minSubArrayLen(15, [1,2,3,4,5]))   # Output: 5

File: MINIMUM-SIZE-SUBARRAY-SUM/msss.go
package main

import "math"

func minSubArrayLen(target int, nums []int) int {
    left, sum := 0, 0
    minLen := math.MaxInt32

    for right := 0; right < len(nums); right++ {
        sum += nums[right]
        for sum >= target {
            if right-left+1 < minLen {
                minLen = right - left + 1
            }
            sum -= nums[left]
            left++
        }
    }
    if minLen == math.MaxInt32 {
        return 0
    }
    return minLen
}
